src/app/views/datasets-new/DatasetsController.jsx

```diff
@@ -0,0 +1,168 @@
+import { push } from 'react-router-redux';
+import PropTypes from 'prop-types';
+
+import datasets from '../../utilities/api-clients/datasets';
+import notifications from '../../utilities/notifications';
+import url from '../../utilities/url';
+import log, {eventTypes} from '../../utilities/log';
+
+import SimpleSelectableList from '../../components/simple-selectable-list/SimpleSelectableList';
+import Input from '../../components/Input';
+
+const propTypes = {
+    dispatch: PropTypes.func.isRequired,
+    location: PropTypes.shape({
+        pathname: PropTypes.string.isRequired
+    }).isRequired
+}
+
+export class DatasetsController extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isFetchingDatasets: false,
+            datasets: [],
+            filteredDatasets: [],
+            searchTerm: ""
+        }
+
+    }
+
+    componentWillMount() {
+        return this.getAllDatasets();
+    }
+
+    getAllDatasets() {
+        this.setState({isFetchingDatasets: true});
+        return datasets.getAll().then(datasets => {
+            this.setState({
+                isFetchingDatasets: false, 
+                datasets: this.mapDatasetsToState(datasets.items)
+            })
+        }).catch(error => {
+            switch (error.status) {
+                case(403):{
+                    const notification = {
+                        "type": "neutral",
+                        "message": "You do not permission to view submitted datasets.",
+                        isDismissable: true
+                    }
+                    notifications.add(notification)
+                    break;
+                }
+                case(404):{
+                    const notification = {
+                        "type": "warning",
+                        "message": "No API route available to get all submitted datasets",
+                        isDismissable: true
+                    }
+                    notifications.add(notification)
+                    break;
+                }
+                case("RESPONSE_ERR"):{
+                    const notification = {
+                        "type": "warning",
+                        "message": "An error's occurred whilst trying to get the submitted datasets.",
+                        isDismissable: true
+                    }
+                    notifications.add(notification)
+                    break;
+                }
+                case("FETCH_ERR"): {
+                    const notification = {
+                        type: "warning",
+                        message: "There's been a network error whilst trying to get the submitted datasets. Please check you internet connection and try again in a few moments.",
+                        isDismissable: true
+                    }
+                    notifications.add(notification);
+                    break;
+                }
+                case("UNEXPECTED_ERR"): {
```

Comment:
is this needed as is identical to default.
(this comment is the same for the 6 additional instances)